### PR TITLE
Add attachments

### PR DESCRIPTION
**Pull request by allanmayanei1**
### Description

Add support attachments the file

#### Related issues

- connect to #185 

### Checklist

- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

File: src/http.js [hunk 1]
```diff
@@ -52,9 +53,14 @@ class HttpClient {
...
     var header;
+    var attachments = []
     var mergeOptions = ['headers'];
+    if (exoptions) {
+      var attachments = exoptions.attachments || [];
+      
+    }
 
-    if (typeof data === 'string') {
+    if (typeof data === 'string' && attachments.lemgth === 0) {
```
Comment (raymondfeng) [inline src/http.js]: Typo, it should be `attachments.length`.

File: src/http.js [hunk 2]
```diff
@@ -52,9 +53,14 @@ class HttpClient {
     };
     var attr;
     var header;
+    var attachments = []
     var mergeOptions = ['headers'];
+    if (exoptions) {
+      var attachments = exoptions.attachments || [];
```
Comment (raymondfeng) [inline src/http.js]: Please remove `var`.

File: src/http.js [hunk 3]
```diff
@@ -52,9 +53,13 @@ class HttpClient {
...
     var attr;
     var header;
+    var attachments = []
     var mergeOptions = ['headers'];
+    if (exoptions) {
+      attachments = exoptions.attachments || [];
+    }
 
-    if (typeof data === 'string') {
+    if (typeof data === 'string' && attachments.lemgth) {
```
Comment (raymondfeng) [inline src/http.js]: `attachments.lemgth` => `attachments.length === 0`?

Comment (allanmayanei1) [reply src/http.js]: I did not understand what the correction would be

Comment (raymondfeng) [reply src/http.js]: Please note `length` is correct instead of `lemgth`.

Comment (raymondfeng) [reply src/http.js]: I assume you are testing if the attachments array is empty.

Comment (allanmayanei1) [reply src/http.js]: silly mistake, now it goes

**Comment by raymondfeng** [comment]
@allanmayanei1 It would be good to add a test case for this new feature.

**Comment by raymondfeng** [comment]
@allanmayanei1 The changes look good now. Would you please squash all commits into one?

**Comment by allanmayanei1** [comment]
> @allanmayanei1 The changes look good now. Would you please squash all commits into one?

changes will rise when ?

**Comment by raymondfeng** [comment]
@allanmayanei1 I have cleaned up the commit history in your repo.

**Comment by allanmayanei1** [comment]
@raymondfeng Thank you, something is missing?

**Comment by raymondfeng** [comment]
Merged and published. Thanks for the contribution.

**Comment by allanmayanei1** [comment]
I thank you

**Pull request merged**